dna-src/zomes/graphql/code/src/schema/post.rs

```diff
@@ -23,21 +28,57 @@ pub struct Post {
 
 #[derive(Serialize, Deserialize, Debug, Clone, DefaultJson)]
 pub struct PostEntry {
-	pub title: String,
+    pub title: String,
     pub details: String,
     pub post_type: String,
     pub creator: Address,
     pub announcement: bool,
     pub timestamp: String,
+    pub base: String,
 }
 
 impl Post {
 	fn retrieve_entry(&self) -> ZomeApiResult<PostEntry> {
-		let id: String = self.id.clone().into();
-		let result = JsonString::from(call_cached("posts", "get_post", json!({"address": id}).into())?);
+		let result = JsonString::from(call_cached("posts", "get_post", 
+			json!({
+				"address": self.id.to_string()
+			}).into())?
+		);
 		let post_entry = PostEntry::try_from(result)?;
 		Ok(post_entry)
 	}
+
+	fn get_comments(&self) -> ZomeApiResult<Vec<Comment>> {
+		let result = call_cached("comments", "get_comments",
+			json!({
+				"base": self.id.to_string()
+			}).into()
+		)?;
+
+	    Ok(
+	    	result.as_array()
+	    	.ok_or(ZomeApiError::Internal(
+	    		format!("Could not parse get comments response: {}", result)
+	    	))?
+	    	.to_vec()
+	    	.into_iter()
+	    	.map(|id| Comment{
+	    		id: id.as_str().unwrap().to_string().into(),
+	    	}).collect()
+	    )
+	}
+
+	fn get_commenters(&self) -> ZomeApiResult<Vec<Person>> {
+		Ok(self.get_comments()?
+			.iter()
+			.map(|comment| {
+				let id: String = comment.retrieve_entry().unwrap().creator;
```

Comment:
ah! I tried something like this, but I didn't know I also had to unwrap it, and also wasn't sure if it was kosher for retrieve_entry to be a public method. thought maybe there was a good reason it wasn't.

Reply:
Also I should really handle this error properly rather than unwrapping but anyway

Reply:
I thin

> All looks good and works great. Only thing missing is it doesn't return the posts sorted from most recent to oldest.

I think I will make another PR for sorting. I think it needs some more discussion